### PR TITLE
Source Chargebee: Declare Unexpected Fields

### PR DESCRIPTION
**Pull request by pnilan**
## What
- Adds missing fields that are documented by the API documentation to
- Updates CustomTransformationComponent to remove original custom field from record
- Bumps connector version to 0.4.2
- Sets QL to 400 to reflect certification
- Closes https://github.com/airbytehq/airbyte-internal-issues/issues/6362

### CODE REVIEW

File: airbyte-integrations/connectors/source-chargebee/source_chargebee/schemas/transaction.json [hunk 1]
```diff
@@ -66,9 +66,6 @@
     "error_code": {
       "type": ["string", "null"]
     },
-    "error-text": {
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-chargebee/source_chargebee/schemas/transaction.json:69]: Why we remove this field? will this be a breaking change?

File: docs/integrations/sources/chargebee.md [hunk 2]
```diff
@@ -99,7 +99,8 @@ The Chargebee connector should not run into [Chargebee API](https://apidocs.char
 
 | Version | Date       | Pull Request                                             | Subject                                                                                             |
 | :------ | :--------- | :------------------------------------------------------- | :-------------------------------------------------------------------------------------------------- |
-| 0.4.1   | 2024-03-03 | [35509](https://github.com/airbytehq/airbyte/pull/35509) | Updates CDK version to latest (0.67.1), updates `site_migration_detail` record filtering                                                   |
+| 0.4.2   | 2024-03-14 | [36037](https://github.com/airbytehq/airbyte/pull/36037) | Adds fields to stream schemas                                                                       |
```
Comment (bazarnov) [inline docs/integrations/sources/chargebee.md:102]: nit: Extended the schemas with missing fields for: x, y, z streams

File: airbyte-integrations/connectors/source-chargebee/source_chargebee/components.py [hunk 3]
```diff
@@ -47,7 +47,16 @@ def transform(
...
+        record["custom_fields"] = []
+        keys_to_remove = []
+        for k, v in record.items():
+            if k.startswith("cf_"):
+                record["custom_fields"].append({"name": k, "value": v})
+                keys_to_remove.append(k)
+
+        for k in keys_to_remove:
+            record.pop(k, None)
+
```
Comment (artem1205) [inline airbyte-integrations/connectors/source-chargebee/source_chargebee/components.py:59]:
```suggestion
        record["custom_fields"] = [{"name": k, "value": record.pop(k)} for k in record.copy() if k.startswith("cf_")]
```

**Review by maxi297 — APPROVED** [review]
Given API Sources is fine with this change, I'm also fine with it. In order words, conditional approval to API Sources approval

File: airbyte-integrations/connectors/source-chargebee/source_chargebee/schemas/transaction.json (hunk 1 above)
Comment (pnilan) [reply airbyte-integrations/connectors/source-chargebee/source_chargebee/schemas/transaction.json:69]: Exactly. Both `error-text` and `error_text` existed in the schema. I removed the incorrect one.

File: airbyte-integrations/connectors/source-chargebee/source_chargebee/components.py (hunk 3 above)
Comment (pnilan) [reply airbyte-integrations/connectors/source-chargebee/source_chargebee/components.py:59]: Great call. Thanks!

Comment (pnilan) [reply airbyte-integrations/connectors/source-chargebee/source_chargebee/components.py]: Added.

File: docs/integrations/sources/chargebee.md (hunk 2 above)
Comment (pnilan) [reply docs/integrations/sources/chargebee.md]: Added

**Review by bazarnov — APPROVED** [review]

**Review by artem1205 — APPROVED** [review]

**Pull request merged**